### PR TITLE
Rely on Python 2.5+ context managers

### PR DESCRIPTION
**Pull request by ekohl**
In Python 2.5 context managers were added, which can help ensure open files are closed. EL 5 shipped with Python 2.4, but EL6 has Python 2.6 meaning we can now rely on this shiny new feature.

### CODE REVIEW

**Comment by ekohl** [comment]
I think it's pulling in pulp 2 in the containers

File: src/katello/repos.py [hunk 1]
```diff
@@ -68,22 +68,15 @@ def remove_cache():
...
-            if cache_file is not None:
-                cache_file.close()
+            with open(ENABLED_REPOS_CACHE_FILE, 'r') as cache_file:
+                return self.data() == json.loads(cache_file.read())
+        except ValueError:
+            return False
+        except IOError:
+            error = sys.exc_info()[1]  # backward and forward compatible way to get the exception
+            if error.errno == errno.ENOENT:
+                return False
```
Comment (evgeni) [inline src/katello/repos.py:79]:
```suggestion
        except IOError as exc:
            if exc.errno == errno.ENOENT:
                return False
```

Should work on 2.7+

Comment (evgeni) [reply src/katello/repos.py:79]: What I don't get is why this is special-cased at all -- if there was an IOError, the cache is not valid.

Comment (evgeni) [reply src/katello/repos.py:79]: Actually, if there was *any* error/exception, it's not valid. The only difference the code right now makes is returning `False` vs `None`, but nobody cares about that.

Comment (ekohl) [reply src/katello/repos.py:76]: I wasn't quite sure about it and needing Python 2.6 at the time. Did we completely drop EL6 support, including in downstream?

Comment (ekohl) [reply src/katello/repos.py]: Updated to simplify. How about this version?

Comment (evgeni) [reply src/katello/repos.py]: RHEL6 is dead-dead-dead soon: https://access.redhat.com/articles/4665701
And those changes won't land downstream before that anyway.

Comment (ekohl) [reply src/katello/repos.py]: What's the soon part? I see since July 1st it's in a "we provide you with docs so you can upgrade" phase.

**Pull request merged**